sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -367,6 +398,32 @@ private Mono<ShouldRetryResult> shouldRetryOnBackendServiceUnavailableAsync(
         return Mono.just(ShouldRetryResult.retryAfter(Duration.ZERO));
     }
 
+    private Mono<ShouldRetryResult> shouldRetryOnRequestTimeout(
```

Comment:
is the request timeout here only for direct mode? what about the timeout for gateway mode?

Reply:
Good catch - I've wired the circuit breaking flow for 408 / `GATEWAY_ENDPOINT_READ_TIMEOUT` as well.